modules/standard/Path.chpl

```diff
@@ -457,4 +457,78 @@ proc file.realPath(): string throws {
      return ("", name);
    }
  }
+
+/* Returns the argument.  If the argument begins with `~` or `~<user>`, replace
+   that substring with that user's home directory, otherwise returns the
+   argument unchanged.
+
+   :arg path: a file or directory, which may or may not include `~` or `~<user>`
+   :type path: `string`
+
+   :return: the argument, replacing any `~` or `~<user>` with the user's home
+                directory
+   :rtype: `string`
+*/
+proc expandUser(path: string) : string
```

Comment:
As a heads up, I spent a little effort yesterday to get this file organized alphabetically.  You are under no obligation to do so yourself, but note that I will move the function myself later if you don't (I would place between the implementation for dirname and file.getParentName)